### PR TITLE
fix PSR-7 contract violation on MessageInterface::getHeaders()

### PR DESCRIPTION
**Pull request by 1ma**
Resubmission of fix for issue 2091 (https://github.com/slimphp/Slim/issues/2091).

### CODE REVIEW

File: src/Headers.php [hunk 1]
```diff
@@ -53,7 +53,7 @@ public static function createFromGlobals(array $globals)
             $key = strtoupper($key);
             if (isset(static::$special[$key]) || strpos($key, 'HTTP_') === 0) {
                 if ($key !== 'HTTP_CONTENT_LENGTH') {
```
Comment (akrabat) [inline src/Headers.php]: Anyone know why we treat `HTTP_CONTENT_LENGTH` differently?

Comment (1ma) [inline src/Headers.php]: My guess is due to this old bug https://bugs.php.net/bug.php?id=66606
`HTTP_CONTENT_LENGTH` (and `HTTP_CONTENT_TYPE`, which used to be in this `if` but was removed) should just not exist, the correct FastCGI header names are `CONTENT_LENGTH` and `CONTENT_TYPE`.

**Comment by akrabat** [comment]
Question: Should we just use `getallheaders()` and ignore `$_SERVER` for headers?

**Comment by 1ma** [comment]
It fits the bill, but it's from the Apache module and not widely available... The [manual](http://php.net/manual/en/function.getallheaders.php) says otherwise, but I did a quick test under nginx+fpm7.1 and it was indeed unavailable.

**Comment by akrabat** [comment]
okay. that's a good reason!

File: src/Headers.php (hunk 1 above)
Comment (akrabat) [inline src/Headers.php]: Do we know which version of PHP that was fixed in?

Comment (akrabat) [inline src/Headers.php]: To answer my own question, it was 5.6.13: https://github.com/php/php-src/blob/PHP-5.6/NEWS#L954

**Pull request merged**